# **US - Baby Names**

### **Step 1. Import the necessary libraries**

In [1]:
import pandas as pd
import numpy as np

### **Step 3. Assign it to a variable called baby_names**

In [2]:
url = 'https://raw.githubusercontent.com/guipsamora/pandas_exercises/master/06_Stats/US_Baby_Names/US_Baby_Names_right.csv'
baby_names = pd.read_csv(url)

In [3]:
baby_names.head()

Unnamed: 0     Id     Name  Year Gender State  Count
0       11349  11350     Emma  2004      F    AK     62
1       11350  11351  Madison  2004      F    AK     48
2       11351  11352   Hannah  2004      F    AK     46
3       11352  11353    Grace  2004      F    AK     44
4       11353  11354    Emily  2004      F    AK     41

### **Step 4. See the first 10 entries**

In [4]:
baby_names.head(10)

Unnamed: 0     Id      Name  Year Gender State  Count
0       11349  11350      Emma  2004      F    AK     62
1       11350  11351   Madison  2004      F    AK     48
2       11351  11352    Hannah  2004      F    AK     46
3       11352  11353     Grace  2004      F    AK     44
4       11353  11354     Emily  2004      F    AK     41
5       11354  11355   Abigail  2004      F    AK     37
6       11355  11356    Olivia  2004      F    AK     33
7       11356  11357  Isabella  2004      F    AK     30
8       11357  11358    Alyssa  2004      F    AK     29
9       11358  11359    Sophia  2004      F    AK     28

### **Step 5. Delete the column 'Unnamed: 0' and 'Id'**

In [5]:
#try this new method of deleting columns
del baby_names['Unnamed: 0']
del baby_names['Id']

In [6]:
baby_names.head()

Name  Year Gender State  Count
0     Emma  2004      F    AK     62
1  Madison  2004      F    AK     48
2   Hannah  2004      F    AK     46
3    Grace  2004      F    AK     44
4    Emily  2004      F    AK     41

### **Step 6. Is there more male or female names in the dataset?**

In [7]:
baby_names['Gender'].value_counts()

F    558846
M    457549
Name: Gender, dtype: int64

There is more female names

<div class="alert alert-block alert-danger">
    This is where my analysis went wrong, I was looking by year and should have not included the year

### **Step 7. Group the dataset by name and assign to names**

In [71]:
#one thing I don't like about groupby, is that to view the data I have to use an aggregation
names = baby_names.groupby('Name').sum()

In [93]:
names = names.sort_values('Count', ascending = False)
del names['Year']

### **Step 8. How many different names exist in the dataset?**

In [94]:
names.info()

<class 'pandas.core.frame.DataFrame'>
Index: 17632 entries, Jacob to Nita
Data columns (total 1 columns):
 #   Column  Non-Null Count  Dtype
---  ------  --------------  -----
 0   Count   17632 non-null  int64
dtypes: int64(1)
memory usage: 275.5+ KB


There are 17,632 names

In [95]:
baby_names['Name'].nunique() #this just confirms

17632

### **Step 9. What is the name with most occurrences?**

In [96]:
names[names['Count'] == names['Count'].max()]

Count
Name         
Jacob  242874

Jacob is the name with the most occurrences

### **Step 10. How many different names have the least occurrences?**

In [97]:
names[names['Count'] == 5]

Count
Name           
Oliverio      5
Daelen        5
Odell         5
Obi           5
Rehana        5
...         ...
Aiyla         5
Jamela        5
Jamera        5
Aivy          5
Nita          5

[2578 rows x 1 columns]

2,578 names with an occurrence of 5

### **Step 11. What is the median name occurrence?**

In [98]:
names[names['Count'] == names['Count'].median()]


Count
Name             
Aveline        49
Nassir         49
Deante         49
Ameir          49
Jelissa        49
...           ...
Alara          49
Dahlila        49
Maisha         49
Sanjuanita     49
Kaio           49

[66 rows x 1 columns]

### **Step 12. What is the standard deviation of names?**

In [99]:
names.std()

Count    11006.069468
dtype: float64

### **Step 13. Get a summary with the mean, min, max, std and quartiles**

In [100]:
names.describe()

Count
count   17632.000000
mean     2008.932169
std     11006.069468
min         5.000000
25%        11.000000
50%        49.000000
75%       337.000000
max    242874.000000